internals/daemon/api_exec.go

```diff
@@ -73,13 +75,29 @@ func v1PostExec(c *Command, req *http.Request, _ *userState) Response {
 		return statusBadRequest("%v", err)
 	}
 
+	// Inherit the pebble daemon environment.
+	environment := make(map[string]string)
+	for _, kv := range os.Environ() {
+		parts := strings.SplitN(kv, "=", 2)
+		key := parts[0]
+		val := ""
+		if len(parts) == 2 {
+			val = parts[1]
+		}
+		environment[key] = val
```

Comment:
Can we please move this too its own function in osutil:
```
func Environ() map[string]string
```
I believe we use similar logic in at least one other place, while running the service command.

Reply:
Yeah, `cmd_exec.go` does it too.

Reply:
I added `osutil.Environ()` in https://github.com/canonical/pebble/pull/234/commits/4487bade960ba4970ee079ea5e77b1c2578bf914, and used it in `cmdstate.Exec` in https://github.com/canonical/pebble/pull/234/commits/e5d80308379917cc14ccf078cc486fcbffff810e.

We are doing a similar thing in [`cmd_exec.go`](https://github.com/canonical/pebble/blob/48d20dc71efb14ed52fdf4430c50f2b39a7676af/internals/cli/cmd_exec.go#L94-L102), but that is on a variable called `cmd.Env` not on `os.Environ()`. To support that, I think I will need to change the signature of `osutil.Environ` to the following:
```
func Environ(env []string) map[string]string
```

I will keep this open. Let me know what you think I should do.

Reply:
Yeah, looking more closely there I don't think we should use it there. I thought it was coming from `os.Environ` but it's for the command-line args (whose syntax just happens to be the same).